server/region_handler.go

```diff
@@ -694,6 +740,139 @@ func (t *regionHandlerTool) getMvccByStartTs(startTS uint64, startKey, endKey []
 	}
 }
 
+func (t *regionHandlerTool) getMvccByIdxValue(idx table.Index, values url.Values, idxCols []*model.ColumnInfo, handleStr string) (*kvrpcpb.MvccGetByKeyResponse, error) {
+	idxRow, err := t.formValue2DatumRow(values, idxCols)
+	if err != nil {
+		return nil, err
+	}
+	handle, err := strconv.ParseInt(handleStr, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	encodeKey, _, err := idx.GenIndexKey(idxRow, handle)
+	if err != nil {
+		return nil, err
+	}
+	keyLocation, err := t.regionCache.LocateKey(t.bo, encodeKey)
+	if err != nil {
+		return nil, err
+	}
+
+	tikvReq := &tikvrpc.Request{
+		Type: tikvrpc.CmdMvccGetByKey,
+		MvccGetByKey: &kvrpcpb.MvccGetByKeyRequest{
+			Key: encodeKey,
+		},
+	}
+	kvResp, err := t.store.SendReq(t.bo, tikvReq, keyLocation.Region, time.Minute)
+	log.Info(string(encodeKey), keyLocation.Region, string(keyLocation.StartKey), string(keyLocation.EndKey), kvResp, err)
+
+	if err != nil {
+		return nil, err
+	}
+	return kvResp.MvccGetByKey, nil
+}
+
+func (t *regionHandlerTool) formValue2DatumRow(values url.Values, idxCols []*model.ColumnInfo) ([]types.Datum, error) {
+	data := make([]types.Datum, len(idxCols))
+	for i, col := range idxCols {
+		var d types.Datum
+
+		vals, ok := values[col.Name.String()]
+		if !ok {
+			return nil, errors.BadRequestf("Missing value for index column %s.", col.Name.String())
+		}
+
+		val := vals[0]
+		if len(val) > 0 {
```

Comment:
what's the difference between `NULL` and empty string?

Reply:
Done.